model/src/order.rs

```diff
@@ -202,19 +192,18 @@ impl OrderCreation {
 // Intended to be used by tests that need signed orders.
 impl OrderCreation {}
 
-// See https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2Encoding.sol
 impl OrderCreation {
+    // See https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2Encoding.sol
     pub const ORDER_TYPE_HASH: [u8; 32] =
         hex!("b2b38b9dcbdeb41f7ad71dea9aed79fb47f7bbc3436576fe994b43d5b16ecdec");
-
     // keccak256("sell")
     const ORDER_KIND_SELL: [u8; 32] =
         hex!("f3b277728b3fee749481eb3e0b3b48980dbbab78658fc419025cb16eee346775");
     // keccak256("buy")
     const ORDER_KIND_BUY: [u8; 32] =
         hex!("6ed88e868af0a1983e3886d5f3e95a2fafbd6c3450bc229e27342283dc429ccc");
 
-    fn order_digest(&self) -> [u8; 32] {
+    pub fn digest(&self) -> [u8; 32] {
```

Comment:
Given that digest is shared among any struct that has a Signature field, I would think we should still have a trait Signing which declares the digest function signature as a mandatory method that needs to be implemented.